acceptance/tests/ticket_3360_allow_duplicate_csr_with_option_set.rb

```diff
@@ -33,18 +41,19 @@
 
   new_certs = {}
   new_cert_list.stdout.each_line do |line|
-    if line =~ /^\+ (\S+) \((.+)\)$/
+    if line =~ /^\+ \"(\S+)\" \(?(.+)\)?$/ && agents_with_cert_name[$1]
       new_certs[$1] = $2
       puts "new cert: #{$1} #{$2}"
     end
   end
 
-  step "Verify the certs have changed"
+  assert_equal(agents.count, new_certs.count,
+               "Expected new number of agent csrs on master to equal number of agents")
+
   # using the agent name as the key may cause errors;
   # agent name from cfg file is likely to have short name
   # where certs might be signed with long names.
   old_certs.each_key { |key|
-    next if key.include? master # skip the masters cert, only care about agents
-    assert_not_equal(old_certs[key], new_certs[key], "Expected #{key} to have a changed key")
+    refute_equal(old_certs[key], new_certs[key], "Expected #{key} to have a changed key")
```

Comment:
Why aren't we skipping the master's cert (or are you wanting to test the agent on the master)?

Why is `assert_not_equal` not good?